adapter.go

```diff
@@ -3,6 +3,31 @@ package bluetooth
 // Set this to true to print debug messages, for example for unknown events.
 const debug = false
 
+// AdapterState represents the state of the adaptor.
+type AdapterState int
+
+const (
+	// AdapterStatePoweredOff is the state of the adaptor when it is powered off.
+	AdapterStatePoweredOff = AdapterState(iota)
+	// AdapterStatePoweredOn is the state of the adaptor when it is powered on.
+	AdapterStatePoweredOn
+	// AdapterStateUnknown is the state of the adaptor when it is unknown.
+	AdapterStateUnknown
```

Comment:
Is it actually possible for the state to be unknown, and under what circumstances can this happen?

If possible, I'd rather avoid this condition as it's just another special case.

If there are only two states, we can just use a boolean (`enabled bool` as a parameter for example).

Reply:
They vary between OS's. 

The MacOS implementation does define an unknown state along with [a few others ](https://developer.apple.com/documentation/corebluetooth/cbmanagerstate). On the linux side the [`Powered`](https://git.kernel.org/pub/scm/bluetooth/bluez.git/tree/doc/adapter-api.txt) state that this PR watches only has `on` or `off`. 

The `Unknown` was just an attempt to do 'something' with the extra ones that Core Bluetooth throws at us. I wasn't sure if the other states are merely transient and the adapter will always eventually fall into an `on | off` state or if there is a chance the states like [`CBManagerStateUnsupported`](https://developer.apple.com/documentation/corebluetooth/cbmanagerstate/cbmanagerstateunsupported?language=objc) may be a permanent latch.